node/tests/isolate_mock.cpp

```diff
@@ -115,12 +126,81 @@ struct spawn_slot_t:
         upstream =  upstream.send<protocol::chunk>("CHUNK1");
         sleep(1);
         upstream = upstream.send<protocol::chunk>("CHUNK2");
+        sleep(1);
+        upstream = upstream.send<protocol::chunk>("CHUNK3");
+        sleep(1);
         upstream.send<protocol::choke>();
         COCAINE_LOG_INFO(log, "spawn. done");
         return boost::optional<std::shared_ptr<dispatch_type>>(std::make_shared<spawned_dispatch_t>());
     }
 };
 
+struct metrics_slot_t:
```

Comment:
In general it's now optional to create a separate slot - you can use a function receiving headers, tuple_type, upstream_type, but it's up to you.